packages/react-priority-overflow/src/components/Overflow.tsx

```diff
@@ -31,43 +21,52 @@ const useStyles = makeStyles({
   },
 });
 
-export interface OverflowProps
-  extends React.HTMLAttributes<HTMLDivElement>,
-    Pick<ObserveOptions, 'overflowAxis' | 'overflowDirection' | 'padding' | 'minimumVisible'> {}
+/**
+ * Overflow Props
+ */
+export type OverflowProps = Partial<
+  Pick<ObserveOptions, 'overflowAxis' | 'overflowDirection' | 'padding' | 'minimumVisible'>
+> & {
+  children: React.ReactElement;
+};
 
-export const Overflow = React.forwardRef<HTMLDivElement, OverflowProps>((props, ref) => {
-  const { overflowAxis = 'horizontal', overflowDirection, padding, minimumVisible, ...rest } = props;
+/**
+ * Provides an OverflowContext for OverflowItem descendants.
+ */
+export const Overflow = React.forwardRef((props: OverflowProps, ref) => {
   const styles = useStyles();
 
+  const { children, minimumVisible, overflowAxis = 'horizontal', overflowDirection, padding } = props;
+
   const [hasOverflow, setHasOverflow] = React.useState(false);
   const [itemVisibility, setItemVisibility] = React.useState<Record<string, boolean>>({});
   const [groupVisibility, setGroupVisibility] = React.useState<Record<string, OverflowGroupState>>({});
 
-  const updateItemVisibility: OnUpdateOverflow = ({
-    visibleItems,
-    invisibleItems,
-    groupVisibility: managerGroupVisibility,
-  }) => {
-    setHasOverflow(invisibleItems.length > 0);
+  // useOverflowContainer wraps this method in a useEventCallback.
+  // TODO: Do we need a useEventCallback here too?
+  const update: OnUpdateOverflow = data => {
+    setHasOverflow(() => data.invisibleItems.length > 0);
     setItemVisibility(() => {
       const newState: Record<string, boolean> = {};
-      visibleItems.forEach(x => (newState[x.id] = true));
-      invisibleItems.forEach(x => (newState[x.id] = false));
+      data.visibleItems.forEach(x => (newState[x.id] = true));
+      data.invisibleItems.forEach(x => (newState[x.id] = false));
       return newState;
     });
-
-    setGroupVisibility(managerGroupVisibility);
+    setGroupVisibility(data.groupVisibility);
   };
 
-  const { containerRef, registerItem, updateOverflow } = useOverflowContainer<HTMLDivElement>(updateItemVisibility, {
+  const { containerRef, registerItem, updateOverflow } = useOverflowContainer(update, {
     overflowDirection,
     overflowAxis,
     padding,
     minimumVisible,
     onUpdateItemVisibility: updateVisibilityAttribute,
   });
 
+  const child = React.Children.only(children);
   const mergedRef = useMergedRefs(containerRef, ref);
+  const mergedClassName = mergeClasses(styles.overflowMenu, styles.overflowingItems, child.props.className);
+  const clonedChild = React.cloneElement(child, { ref: mergedRef, className: mergedClassName });
```

Comment:
```suggestion
const clonedChild = applyTriggerPropsToChildren(children, {
    ref: useMergedRefs(containerRef, ref),
    clasName: mergeClasses(styles.overflowMenu, styles.overflowingItems, child.props.className),
  });
```

Let's be consistent in how we clone children in the repo. You can import this helper from react-utilities

Reply:
Updated directly to add the include and the className property.